### PR TITLE
(#949) DateOf now parses also date without time

### PR DESCRIPTION
**Pull request by victornoel**
This is for #949.

Added a test for the issue and implemented a solution based on https://stackoverflow.com/a/39685202/9046139

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1093](https://github.com/yegor256/cactoos/issues/1093) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1093](https://github.com/yegor256/cactoos/issues/1093) is assigned to @atapin[/z](https://www.0crat.com/u/atapin), here is [why](https://www.0crat.com/footprint/C63314D6Z/eb3c3814-62ac-425a-a503-e10d5aaede20); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/time/DateOfTest.java [hunk 1]
```diff
@@ -73,6 +73,24 @@ public final void testParsingCustomFormattedStringToDate() {
         ).affirm();
     }
 
+    @Test
+    public final void testParsingCustomFormattedStringWithoutTimeToDate() {
+        new Assertion<>(
+            "Can't parse a Date with custom format.",
```
Comment (atapin) [inline src/test/java/org/cactoos/time/DateOfTest.java]: I'm confused with the negative clause of the assertion. I see that it's kinda an error message stub for a failed test case, but we have description of an intent in the other tests. @llorllale do we use `Must/should do something` statement or `Can't do something`?

Comment (victornoel) [reply src/test/java/org/cactoos/time/DateOfTest.java]: @atapin good point, I always struggle with that. Usually I use the "must" form, but here I copy-pasted it from another tests and didn't change the message. Let's see what @llorllale says but I propose to change the wording with "must" (as in `Assertion` javadoc).

Comment (llorllale) [reply src/test/java/org/cactoos/time/DateOfTest.java]: @atapin you are 100% correct, ie use a positive clause starting with a modal verb like "must"

Comment (atapin) [reply src/test/java/org/cactoos/time/DateOfTest.java]: @victornoel can we fix unrelated test cases in the scope of the current PR?

**Comment by victornoel** [comment]
@atapin I've corrected the tests message, all of them while I was at it.

**Comment by atapin** [comment]
@llorllale ping

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @atapin[/z](https://www.0crat.com/u/atapin), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1093](https://github.com/yegor256/cactoos/issues/1093) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0pdd quality acceptable

@atapin has mentioned @llorllale before writing comment in [PR](https://github.com/yegor256/cactoos/pull/1093#discussion_r271188920)

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/yegor256/cactoos/issues/1093#issuecomment-480564184))

@sereshqua There is no quality review for [#1093](https://github.com/yegor256/cactoos/issues/1093), no performer

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @atapin[/z](https://www.0crat.com/u/atapin)

**Pull request merged**